### PR TITLE
Data preservation

### PR DESCRIPTION
**Pull request by gdumani**
In this milestone, data preservation methods were added:
- save and retrieve data for
  - People
  - Books
  - Rentals
- Information is saved in JSON format

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -124,4 +125,57 @@ def list_rentals
...
+
+  def load_books(filename)
+    if File.exist?(filename)
+      @books = JSON.parse(File.read(filename)).map do |b|
+        Book.new(b['title'], b['author'])
+      end
+    else
+      File.write(filename, '[]')
+    end
+  end
```
Comment (mikethreels) [inline app.rb:145]: - [x] If the file is empty the user receives the below error, please make sure that the application works even if the JSON files are empty.

File: app.rb [hunk 2]
```diff
@@ -124,4 +125,57 @@ def list_rentals
...
+          Student.new(p['age'], p['name'], p['parent_permission'])
+          # student.id = p['person_id']
+        elsif p['type'] == 'Teacher'
+          Teacher.new(p['age'], p['name'], p['specialization'])
+        end
+      end
+    else
+      File.write(filename, '[]')
+    end
+  end
```
Comment (mikethreels) [inline app.rb:160]: - [ ] If the file is empty the user receives the below error, please make sure that the application works even if the JSON files are empty.

File: app.rb [hunk 3]
```diff
@@ -124,4 +125,57 @@ def list_rentals
...
+  def load_rentals(filename)
+    if File.exist?(filename)
+      @rentals = JSON.parse(File.read(filename)).map do |r|
+        book = @books.find { |b| b.title == r['book']['title'] }
+        person = @people.find { |p| p.name == r['person']['name'] }
+        Rental.new(r['date'], person, book)
+      end
+    else
+      File.write(filename, '[]')
+    end
```
Comment (mikethreels) [inline app.rb]: - [ ] If the file is empty the user receives the below error, please make sure that the application works even if the JSON files are empty.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job updating the JSON files when closing the application.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (gdumani) [reply app.rb:145]: Error happend when loading rentals to inexistent persons or books
It was fixed by adding a condition to the rental loading method

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @banzaman & @gdumani , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] No linter errors :white_check_mark: 
- [x] Good console UI design :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark: 
- [x] Implemented the changes from the previous reviewer :white_check_mark:    

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**